File: Python/7kyu/ValidatePIN.py
import re


def validate_pin_fifi(pin):
    if len(pin) == 0 or re.search('[^\d]', pin) or int(pin) < 0:
        return False

    return len(pin) == 4 or len(pin) == 6


def validate_pin(pin):
    return len(pin) in (4, 6) and pin.isdigit()


if __name__ == "__main__":
    print(validate_pin(""))

File: CSharp/7kyu/ValidatePIN.cs
using System.Linq;
using System.Text.RegularExpressions;

namespace CSharp._7Kyu
{
    public static partial class Kata7
    {
        public static bool ValidatePin_fifi(string pin)
        {
            int[] nums = { 4, 6 };
            return nums.Contains(pin.Length) && !Regex.Match(pin, @"[^\d]").Success;
        }

        public static bool ValidatePin(string pin)
        {
            return pin.All(p => char.IsDigit(p)) && (pin.Length == 4 || pin.Length == 6);
        }
    }
}